oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyState.java

```diff
@@ -116,7 +155,20 @@ public int count() {
      */
     @NotNull
     String getValue() {
-        return value;
+        return value != null ? value : decompress(this.compressedValue);
+    }
+
+    private String decompress(byte[] value) {
+        try {
+            return new String(compression.getInputStream(new ByteArrayInputStream(value)).readAllBytes());
+        } catch (IOException e) {
+            LOG.error("Failed to decompress property {} value: ", getName(), e);
+            return "\"{}\"";
+        }
+    }
+
+    public byte[] getCompressedValue() {
```

Comment:
This could probably be package protected?

Reply:
1. Can we avoid creating a stream wrapper here, even if this needs changes in compression?
2. Constructing a string from bytes should be done with an explicit character encoding, otherwise it's platform/locale dependant.

Reply:
Done point 2. For 1. I wouldn't add changes in compression interface, only if indeed required.